src/core/src/service_interfaces/StatusHandler.py

```diff
@@ -405,20 +421,31 @@ def __new_installation_summary_json(self, installation_packages_json):
         self.__refresh_installation_reboot_status()
 
         # Compose substatus message
+        maintenance_run_id = self.execution_config.maintenance_run_id if self.execution_config.maintenance_run_id is not None else ''
+        errors = self.__set_errors_json(self.__installation_total_error_count, self.__installation_errors)
+        substatus_message = self.__compose_installation_substatus_msg(activity_id=self.execution_config.activity_id, reboot_status=self.__installation_reboot_status,
+            maintenance_window=self.__maintenance_window_exceeded, not_selected=not_selected_patch_count, excluded=excluded_patch_count,
+            pending=pending_patch_count, installed=installed_patch_count, failed=failed_patch_count,
+            packages=installation_packages_json, start_time=self.execution_config.start_time,
+            last_modified_time=self.env_layer.datetime.timestamp(), maintenance_id=maintenance_run_id, errors=errors)
+
+        return substatus_message
+
+    def __compose_installation_substatus_msg(self, activity_id, reboot_status, maintenance_window, not_selected, excluded, pending, installed, failed, packages, start_time, last_modified_time, maintenance_id, errors):
         return {
-            "installationActivityId": str(self.execution_config.activity_id),
-            "rebootStatus": str(self.__installation_reboot_status),
-            "maintenanceWindowExceeded": self.__maintenance_window_exceeded,
-            "notSelectedPatchCount": not_selected_patch_count,
-            "excludedPatchCount": excluded_patch_count,
-            "pendingPatchCount": pending_patch_count,
-            "installedPatchCount": installed_patch_count,
-            "failedPatchCount": failed_patch_count,
-            "patches": installation_packages_json,
-            "startTime": str(self.execution_config.start_time),
-            "lastModifiedTime": str(self.env_layer.datetime.timestamp()),
-            "maintenanceRunId": str(self.execution_config.maintenance_run_id) if self.execution_config.maintenance_run_id is not None else '',
-            "errors": self.__set_errors_json(self.__installation_total_error_count, self.__installation_errors)
+            "installationActivityId": str(activity_id),
+            "rebootStatus": str(reboot_status),
+            "maintenanceWindowExceeded": maintenance_window,
+            "notSelectedPatchCount": not_selected,
+            "excludedPatchCount": excluded,
+            "pendingPatchCount": pending,
+            "installedPatchCount": installed,
+            "failedPatchCount": failed,
+            "patches": packages,
+            "startTime": str(start_time),
+            "lastModifiedTime": str(last_modified_time),
+            "maintenanceRunId": str(maintenance_id),
+            "errors": errors
```

Comment:
this refactoring was needed at the time when I was recomposing the assessment and installation substatus, then I modified the recompose to be shorter, so the refactor method wasn't apply anymore. I left the changes as it is because i see the benefit of having it extracted out in a method that it can be use in truncation logic again or else where

Reply:
Please refactor it in the PR that first uses this structure of calling - thereby justifying the change in the context of actual use. If no future PR is going to rely on this structure, then the programming principle YAGNI applies - https://en.wikipedia.org/wiki/You_aren%27t_gonna_need_it

When a significantly complex group of changes are going in, it's also important to reduce the surface area for investigation and debugging later if something goes wrong, and also to reduce load on code reviewers (which has been emphasized a few times already).

Please revert this and any large change that's unnecessary to complete the functional aspects of what is trying to be achieved. If a subsequent PR justifies the change, we can review it in that PR.

Reply:
addressed